.github/workflows/unit_test.yml

```diff
@@ -1,68 +1,86 @@
 jobs:
-  build-and-test:
+  run_unit_tests:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - uses: conda-incubator/setup-miniconda@v2.0.0
+    -
+      name: Log in to the Container registry
+      uses: docker/login-action@v1
       with:
-        miniconda-version: 'latest'
-        activate-environment: test-environment
-        python-version: 3.6.10
-        auto-activate-base: false
-        auto-update-conda: true
-        condarc-file: test/condarc.yml
-
-    - name: Use Node JS 16.x
-      uses: actions/setup-node@v2.1.2
-      with:
-        node-version: 16.x
-
-    - name: Install JS dependencies
+        registry: ghcr.io
+        username: ${{ secrets.GHCR_USERNAME }}
+        password: ${{ secrets.GHCR_TOKEN }}
+    -
+      name: Repo checkout
+      uses: actions/checkout@v2
+    -
+      name: Set up environment
       run: |
-        npm ci
-        npm run install-npm
-
-    - name: Install Narrative Application
+        if [ ${{ github.base_ref }} == 'master' ]; then
+            branch=''
+        else
+            branch="-""${{ github.base_ref }}"
+        fi
+        narrative_version=`grep '\"version\":' src/config.json.templ | awk '{print $2}' | sed 's/"//g'`
+        narrative_git_hash=`grep '\"git_commit_hash\":' src/config.json.templ | awk '{print $2}' | sed 's/"//g' | sed 's/,//'`
+        echo "APP_IMAGE_TAG=ghcr.io/${{ github.repository }}${branch}:pr-${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+        echo "APP_VERSION_IMAGE_TAG=ghcr.io/${{ github.repository }}${branch}_version:pr-${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+        echo "BRANCH=${branch}" >> $GITHUB_ENV
+        echo "BUILD_DATE=$(date -u +"%Y-%m-%dT%H:%M:%SZ")" >> $GITHUB_ENV
+        echo "NARRATIVE_GIT_HASH=${narrative_git_hash}" >> $GITHUB_ENV
+        echo "NARRATIVE_VERSION=${narrative_version}" >> $GITHUB_ENV
+        echo "PR=pr-${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+        echo "VCS_REF=$(echo "${{ github.event.pull_request.head.sha }}" | cut -c -7)" >> $GITHUB_ENV
+    -
+      name: Pull docker image
       shell: bash -l {0}
-      run: |
-          bash ./scripts/install_narrative.sh
-          npm run minify
-          sed <src/config.json.templ >src/config.json "s/{{ .Env.CONFIG_ENV }}/dev/"
-          sed -i 's/{{ if ne .Env.CONFIG_ENV "prod" }} true {{- else }} false {{- end }}/true/' src/config.json
-          jupyter notebook --version
-
-    - name: Set up environment vars
-      run: |
-        echo "KBASE_TEST_TOKEN=${{ secrets.NARRATIVE_TEST_TOKEN }}" >> $GITHUB_ENV
-
-    - name: Run Narrative Backend Tests
+      run: docker pull ${{ env.APP_IMAGE_TAG }}
+    -
+      name: Run backend tests
       id: test_backend
       shell: bash -l {0}
-      run: make test-backend
+      run: |
+        mkdir -p ${{ github.workspace }}/output
+        sudo chown nobody ${{ github.workspace }}/output
+        sudo chmod -R 0777 ${{ github.workspace }}/output
```

Comment:
fix the permissions of this directory so that it can be mounted with the docker image and used for the coverage file from the backend tests